### PR TITLE
Introduce a new --dbname-filter flag

### PR DESCRIPTION
**Pull request by kmoppel**
Allows to blend out any activities from DBs that are not of interest - DB size, TPS, running / waiting / blocking queries. To understand when a regex filter is specified I've added a star (*) to the DATABASE column...hopefully this is enough. Looks like that:
Screenshot from 2021-12-12 18-58-08

### CODE REVIEW

**Comment by blogh** [comment]
Hi,

I was thinking about adding more filters, it's great you made this patch !
I will try to have a stab a it this week or next week.

 Thx a lot for the patch.

**Comment by blogh** [comment]
Just tested it; works as expected.

Maybe we should add that the regexp are documented in the following link in a FAQ entry and/or in the man page ?
https://www.postgresql.org/docs/current/functions-matching.html#POSIX-SYNTAX-DETAILS

**Comment by dlax** [comment]
If we want to add more filters, maybe it'd be nicer to have a single `--filter` option that would take a key/value to indicate the object we filter on; e.g. `--filter=dbname:<regex> --filter=username:<regex>`.

**Comment by kmoppel-cognite** [comment]
@blogh Thanks for verifying that it works! But about adding the link to regex docs...probably just overkill...as the people using this tool, especially the more exotic flags, are already very technical people and surely know what a regex means / looks like :)

**Comment by kmoppel-cognite** [comment]
@dlax Sure, not against the idea...but then the architecture for such generic filters probably needs to come from your side.
The hardest part will probably be keeping the SQL templates more or less sane in case of dynamic input...

**Comment by blogh** [comment]
@kmoppel I am not sure about that, but we can wait about the questions and add the FAQ entry then.

**Comment by dlax** [comment]
Added the generic `--filter` flag; can you give it a try?

**Comment by blogh** [comment]
I like the changes :)

**Review by dlax — APPROVED** [review]

**Comment by dlax** [comment]
I've cleaned up the git history.
This looks good to me now so we can merge this I think.
@blogh, what do you think?

(I'll fix the lint issue, due to mypy new version, in the meantime.)

File: README.md [hunk 1]
```diff
@@ -60,8 +60,9 @@ Options
                               2-TRANSACTION, 3-BACKEND
         --min-duration        Don't display queries with smaller than specified
                               duration (in seconds).
-        --dbname-filter=dbname_filter
-                              A regex to show queries only from DBs of interest
+        --filter=FIELD:REGEX  Filter activities with a (case insensitive) regular
```
[the remark this answers is not in the document]
Comment (dlax) [reply README.md:63]: Yes, "(case insensitive)" applies to "regular expression" in this sentence. Or should we reformulate?

[the remark this answers is not in the document]
Comment (blogh) [reply README.md:63]: > Filter activities with a  regular expression applied on selected fields. Known fields are: dbname. The field name is case > insensitive.

would this work ?

**Review by blogh — APPROVED** [review]

**Comment by dlax** [comment]
Merged.
Thank you @kmoppel for the patch and @blogh for the review!

**Pull request merged**